cloudflare/resource_cloudflare_zone.go

```diff
@@ -226,28 +254,19 @@ func flattenMeta(d *schema.ResourceData, meta cloudflare.ZoneMeta) map[string]in
 }
 
 func setRatePlan(client *cloudflare.API, zoneID string, planID string) error {
-	plan, err := getAvailableZonePlan(client, zoneID, planID)
-	if err != nil {
-		return fmt.Errorf("Error fetching plans %s for zone %q: %s", planID, zoneID, err)
-	}
-	log.Printf("[DEBUG] ratePlan = %#v", plan)
-	if _, err := client.ZoneSetPlan(zoneID, *plan); err != nil {
+	if err := client.ZoneSetPlan(zoneID, subscriptionIDOfRatePlans[planID]); err != nil {
 		return fmt.Errorf("Error setting plan %s for zone %q: %s", planID, zoneID, err)
 	}
-	return nil
-}
 
-func getAvailableZonePlan(client *cloudflare.API, zoneID, planID string) (*cloudflare.ZonePlan, error) {
-	plans, err := client.AvailableZonePlans(zoneID)
-	if err != nil {
-		return nil, err
-	}
-	for _, p := range plans {
-		if strings.EqualFold(p.LegacyID, planID) {
-			return &p, nil
-		}
-	}
-	return nil, fmt.Errorf("plan '%s' not found amongst the available plans", planID)
+	// Due to the async delivery of the subscription service, there is
+	// potential that the update is made and we query the endpoint
+	// before it's populated in other systems. This is an arbitary number
+	// and it would be preferrable to work out a better way of polling
+	// for changes without having to do it in every system that sets the
+	// rate plan but it works.
+	time.Sleep(3000 * time.Millisecond)
```

Comment:
@patryk any thoughts on this random wait time? Do you have a better approach or insights here that might allow us to do away with a static sleep?

Reply:
Progressively longer sleep times, as you implemented in newer commit, is good, we can keep it.